### PR TITLE
Fix handling of parseIP error

### PR DESCRIPTION
**Pull request by Dean-Coakley**
Fixes: https://github.com/kubernetes/minikube/issues/8646

### Before changes:
```
$ minikube ip delete
$ minikube start
...
🐳  Preparing Kubernetes v1.18.3 on Docker 19.03.8 ...
```
* In a separate terminal session
```
$ minikube ip
<nil>
$ echo $?
0
```

### Fix
* (Same repro steps)
```
$ minikube ip

💣  Unable to get forwarded endpoint: failed to parse ip

😿  minikube is exiting due to an error. If the above message is not useful, open an issue:
👉  https://github.com/kubernetes/minikube/issues/new/choose
➜  minikube git:(fix-minikube-ip-errhandle) echo $?
70
```

### CODE REVIEW

File: pkg/minikube/driver/endpoint.go [hunk 1]
```diff
@@ -30,6 +32,10 @@ func ControlPlaneEndpoint(cc *config.ClusterConfig, cp *config.Node, driverName
 		port, err := oci.ForwardedPort(cc.Driver, cc.Name, cp.Port)
 		hostname := oci.DefaultBindIPV4
 		ip := net.ParseIP(hostname)
+		if ip == nil {
+			errMsg := fmt.Sprintf("failed to parse ip")
+			err = errors.Wrap(err, errMsg)
```
Comment (Dean-Coakley) [inline pkg/minikube/driver/endpoint.go:37]: * Is Wrap correct here?

* I'd like to add additional context, but from my testing the `ParseIP` fails due to an empty `hostname`. So I'm not sure what I can add that is helpful.

File: pkg/minikube/driver/endpoint.go [hunk 2]
```diff
@@ -30,6 +32,10 @@ func ControlPlaneEndpoint(cc *config.ClusterConfig, cp *config.Node, driverName
 		port, err := oci.ForwardedPort(cc.Driver, cc.Name, cp.Port)
 		hostname := oci.DefaultBindIPV4
 		ip := net.ParseIP(hostname)
+		if ip == nil {
+			errMsg := fmt.Sprintf("failed to parse ip")
```
Comment (medyagh) [inline pkg/minikube/driver/endpoint.go:36]: since the var errMsg is not used anywhere else, consider just putting it in the Wrap

File: pkg/minikube/driver/endpoint.go (hunk 1 above)
Comment (medyagh) [reply pkg/minikube/driver/endpoint.go:37]: instead of error.wraap
you can use fmt.Errorf("failed to parse ip for %s", hostname)

File: pkg/minikube/driver/endpoint.go [hunk 3]
```diff
@@ -43,5 +47,9 @@ func ControlPlaneEndpoint(cc *config.ClusterConfig, cp *config.Node, driverName
 	if cc.KubernetesConfig.APIServerName != constants.APIServerName {
 		hostname = cc.KubernetesConfig.APIServerName
 	}
-	return hostname, net.ParseIP(cp.IP), cp.Port, nil
+	ip := net.ParseIP(cp.IP)
+	if ip == nil {
+		return hostname, ip, cp.Port, fmt.Errorf("failed to parse ip for %s", hostname)
```
Comment (medyagh) [inline pkg/minikube/driver/endpoint.go:52]: here it should be failed to parse ip for %q", cp.IP)

Comment (Dean-Coakley) [reply pkg/minikube/driver/endpoint.go:52]: Sorry!

**Comment by Dean-Coakley** [comment]
(Kept same commit structure, needed to rebase to fix lint test)

**Comment by sharifelgamal** [comment]
@Dean-Coakley can you rebase against master so we can retest? This change looks good and I'd like to merge it soon.

**Comment by medyagh** [comment]
thank you @Dean-Coakley
